preprocess/tokenize.py

```diff
@@ -0,0 +1,48 @@
+from argparse import ArgumentParser
+from os import listdir, remove
+from os.path import splitext, join, isdir, exists
+
+import youtokentome as yttm
+
+from configs import default_tokenizer_config, test_tokenizer_config
+
+data = "data"
+
+
+def tokenize(dataset_path: str, model_path: str, is_test: bool):
+    buffer_path = "text.yttm"
+    if exists(buffer_path):
+        remove(buffer_path)
+
+    for file in listdir(dataset_path):
+        transformed_files_path = join(dataset_path, file)
+        if isdir(transformed_files_path):
+            for transformed_file in listdir(transformed_files_path):
+                file_path = join(transformed_files_path, transformed_file)
+                _, ext = splitext(file_path)
+                if ext == ".cpp":
+                    with open(file_path, "r", encoding="utf8", errors='ignore') as file_:
+                        text = file_.read() + "\n"
+                        with open(buffer_path, "a") as buffer_:
+                            buffer_.write(text)
+
+    tokenizer_config = test_tokenizer_config if is_test else default_tokenizer_config
+    tokenizer = yttm.BPE.train(
+        data="text.yttm",
+        model=model_path,
+        **tokenizer_config.__dict__
```

Comment:
Don't use direct call. There are a special method for converting data class into the dict
https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict